src/vsphere_cpi/spec/unit/cloud/vsphere/vm_config_spec.rb

```diff
@@ -20,6 +20,41 @@ def fake_datastore(name, free_space, mob = nil)
     let(:cluster_picker) { ClusterPicker.new(0, 0) }
     let(:cluster_provider) { nil }
 
+    describe '#upgrade_hw_version?' do
+      context 'when upgrade hardware version is nil (not specified) in both vm_type and global config' do
+        let(:cloud_properties) {{}}
+        let(:vm_type) { VmType.new(datacenter, cloud_properties)}
```

Comment:
definition of vm_type is same as what is in the outer block so you can remove it from all 3 context